src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -684,14 +696,17 @@ private void applyWAL() throws ConnectException {
   }
 
   private void truncateWAL() throws ConnectException {
-    if (!recovered) {
```

Comment:
The previous logic called truncate before recovering the offsets. With the new changes the [offsets are recovered before](https://github.com/confluentinc/kafka-connect-hdfs/blob/a5865bd03872420599c04e76a6985b8fe15f14d2/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L266) the attempt to truncate so `recover` will not have the chance to be false by the time this is reached.

Reply:
Note that `truncateWAL()` is only called in [recover()](https://github.com/confluentinc/kafka-connect-hdfs/blob/a5865bd03872420599c04e76a6985b8fe15f14d2/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L253), which means the recovery will always happen before the call. `wal.truncate()` will be safe to call here without the check since it is always called after [apply()](https://github.com/confluentinc/kafka-connect-hdfs/blob/a5865bd03872420599c04e76a6985b8fe15f14d2/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L262), and this PR also adds file existence checks in the FSWAL class.